manifests/klusterlet/managed/klusterlet-work-clusterrole-execution.yaml

```diff
@@ -26,3 +26,7 @@ rules:
 - apiGroups: ["monitoring.coreos.com"]
   resources: ["servicemonitors"]
   verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+# Allow agent to manage oauth clients
```

Comment:
if you want to use server side apply, also need the patch verb. Also would you add a TODO here that we would need to refactor permission control of work agent to remove this part later?